src/core/utils/CorePlugin.ts

```diff
@@ -20,7 +30,96 @@ export class CorePlugin extends JWPlugin {
+        this._setRange(intent.payload['value']);
+    }
+    /**
+     * Render the `vDocument`.
+     */
+    render(): void {
+        this.vDocument.renderer.render(this.vDocument.editable);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a position in the `VDocument` as a tuple `[VNode, 'before' | 'after']`
+     * (ie. the position is before or after a given `VNode`). The position is
+     * always given on the leaf.
+     *
+     * @param container
+     * @param offset
+     */
+    _getPosition(container: DOMElement | Node, offset: number): VPosition {
+        const containers = this.vDocument.domMap.get(container);
+        const isTextNode = container.nodeType === 3;
+        if (isTextNode) {
+            return this._positionToLeaf({
+                ref: containers[offset] || containers[containers.length - 1],
+                side: offset < containers.length ? 'before' : 'after',
+            });
+        }
+        return this._positionToLeaf({ ref: containers[offset], side: 'before' });
+    }
+    /**
+     * Return a tuple with the range nodes in ltr order (rtl if `reverse` is
+     * true) and updates their `order` property accordingly.
+     *
+     * @param reverse
+     */
+    _orderedRangeNodes(reverse = false): [VNode, VNode] {
+        const rangeNodes = this.vDocument.rangeNodes;
+        const res: [VNode, VNode] = [rangeNodes.start, rangeNodes.end];
+        if (reverse) {
+            res.reverse();
+        }
+        res.forEach((rangeNode: VNode, i: 0 | 1) => {
+            rangeNode.order = i;
+        });
+        return res;
+    }
+    /**
+     * Move the position to its deepest first descendent.
+     *
+     * @param position
+     */
+    _positionToLeaf(position: VPosition): VPosition {
+        while (position.ref.children.length) {
+            position.ref = position.ref.firstChild;
+            position.side = 'before';
+        }
+        return position;
+    }
+    /**
+     * Move the `Range VNodes` to the position in `VDocument` corresponding to
+     * the given DOM Range and trigger a rendering.
+     *
+     * @param range
+     */
+    _setRange(range: Range): void {
+        const startPos: VPosition = this._getPosition(range.startContainer, range.startOffset);
+        const endPos: VPosition = this._getPosition(range.endContainer, range.endOffset);
+        const [first, last] = this._orderedRangeNodes(range['direction'] === 'rtl');
+
+        // Move the range nodes. If the start has to be moved *after* its
+        // reference node, the end should be moved first.
+        if (startPos.side === 'after') {
+            endPos.ref[endPos.side](last);
+            startPos.ref[startPos.side](first);
+        } else {
+            startPos.ref[startPos.side](first);
+            endPos.ref[endPos.side](last);
+        }
```

Comment:
I won't lie: I don't understand anything of what is going on in this function, except that we're supposedly moving the range... :/

Reply:
1. Getting the position where the range nodes should be (before or after which node)
2. Putting it there by calling `before()` or `after()` on the reference node and passing it the correct range node